satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,209 @@
+      >>> local_scene[...].attrs["area"] = area_def
+      >>> local_nwc[...].attrs["area"] = area_def
+
+    Note that the ``ctth`` dataset must contain geolocation metadata, such as
+    set in the ``orbital_parameters`` dataset attribute by many readers.
+    """
+
+    def __init__(self, base_area):
+        """Initialise parallax correction class.
+
+        Args:
+            base_area (pyresample.AreaDefinition): Area for which calculated
+                geolocation will be calculated.
+        """
+        self.base_area = base_area
+
+    def __call__(self, cth_dataset):
+        """Apply parallax correction to dataset.
+
+        Args:
+            cth_dataset: Dataset containing cloud top heights (or other heights
+                to be corrected).
+
+        Returns:
+            pyresample.geometry.SwathDefinition: Swathdefinition with corrected
+                lat/lon coordinates.
+        """
+        return self.corrected_area(cth_dataset)
+
+    def corrected_area(self, cth_dataset):
+        """Corrected area.
+
+        Calculate the corrected SwathDefinition for dataset.
+
+        Returns a parallax corrected swathdefinition of the base area.
+        """
+        area = cth_dataset.area
+        (sat_lon, sat_lat, sat_alt) = get_satpos(cth_dataset)
+        cth_dataset = self._preprocess_cth(cth_dataset)
+        (pixel_lon, pixel_lat) = area.get_lonlats()
+
+        # Pixel coordinates according to parallax correction
+        (corr_lon, corr_lat) = forward_parallax(
+            sat_lon, sat_lat, sat_alt,
+            np.array(pixel_lon), np.array(pixel_lat), np.array(cth_dataset)
+        )
+
+        corr_lon = xr.DataArray(corr_lon)
+        corr_lat = xr.DataArray(corr_lat)
+        corr_area = SwathDefinition(corr_lon, corr_lat)
+
+        # But we are not actually moving pixels, rather we want a
+        # coordinate transformation. With this transformation we approximately
+        # invert the pixel coordinate transformation, giving the lon and lat
+        # where we should retrieve a value for a given pixel.
+        (proj_lon, proj_lat) = self._invert_lonlat(
+                pixel_lon, pixel_lat, corr_area)
+        proj_lon = xr.DataArray(proj_lon)
+        proj_lat = xr.DataArray(proj_lat)
+
+        return SwathDefinition(proj_lon, proj_lat)
+
+    def _preprocess_cth(self, cth_dataset):
+        """To be documented."""
+        units = cth_dataset.units
+        cth = cth_dataset.copy().fillna(0.0)
+        if units == 'm':  # convert to km
+            cth = cth * 1e-3
+        return cth
+
+    def _invert_lonlat(self, pixel_lon, pixel_lat, source_area):
+        """Invert the lon/lat coordinate transformation.
+
+        When a satellite observes a cloud, a reprojection onto the cloud has
+        already happened.
+        """
+        (source_lon, source_lat) = source_area.get_lonlats()
+        resampler = NumpyBilinearResampler(source_area, self.base_area, 50000.0)
```

Comment:
There should be something in pyresample that can do this in a way that works at the antimeridian and the poles.